src/libraries/System.Net.Http/src/System/Net/Http/HttpRequestError.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Net.Http
+{
+    /// <summary>
+    /// Defines error categories representing the reason for <see cref="HttpRequestException"/> or <see cref="HttpIOException"/>.
+    /// </summary>
+    public enum HttpRequestError
+    {
+        /// <summary>
+        /// A generic or unknown error occured.
+        /// </summary>
+        Unknown = 0,
+
+        /// <summary>
+        /// The DNS name resolution failed.
+        /// </summary>
+        NameResolutionError,
+
+        /// <summary>
+        /// A transport-level failure occured while connecting to the remote endpoint.
+        /// </summary>
+        ConnectionError,
+
+        /// <summary>
+        /// An error occured during the TLS handshake.
+        /// </summary>
+        SecureConnectionError,
+
+        /// <summary>
+        /// An HTTP/2 or HTTP/3 protocol error occured.
+        /// </summary>
+        HttpProtocolError,
+
+        /// <summary>
+        /// Extended CONNECT for WebSockets over HTTP/2 is not supported by the peer.
+        /// </summary>
+        ExtendedConnectNotSupported,
+
+        /// <summary>
+        /// Cannot negotiate the HTTP Version requested.
+        /// </summary>
+        VersionNegotiationError,
+
+        /// <summary>
+        /// The authentication failed with the provided credentials.
+        /// </summary>
+        UserAuthenticationError,
+
+        /// <summary>
+        /// An error occured while establishing a connection to the proxy tunnel.
```

Comment:
```suggestion
        /// An error occurred while establishing a proxy tunnel.
```
If this is used more broadly than just when connecting to the proxy.
Same for the `Strings.resx` entry.

Nit: Find-replace `occured` => `occurred`